src/test/java/org/embulk/util/guess/TestTimeFormatGuess.java

```diff
@@ -0,0 +1,134 @@
+                "Fri, 20 Feb 2016 22:00 Z",
+                "Fri, 20 Feb 2016 22:01 -05:00",
+                "Fri, 20 Feb 2016 22:02 +01:00",
+                "Fri, 20 Feb 2016 22:03 +01:00",
+                "Fri, 20 Feb 2016 22:03:03 +00:00",
+                "Fri, 20 Feb 2016 22:04:04 +03:00",
+                "Fri, 20 Feb 2016 22:04 +12:00",
+                "Fri, 20 Feb 2016 22:05 +02:00",
+                "Fri, 20 Feb 2016 22:06 +10",
+                "Fri, 20 Feb 2016 22:06 Z",
+                "Fri, 20 Feb 2016 22:07 +1230");
+    }
+
+    @SuppressWarnings("checkstyle:ParenPad")
+    @Test
+    public void testFormatRfc2822() {
+        // This test is disabled because of https://github.com/jruby/jruby/issues/3702
+        // assertTimeFormat("%a, %d %b %Y %H:%M:%S %Z", "Fri, 20 Feb 2015 14:02:34 PST");
+        assertTimeFormat("%a, %d %b %Y %H:%M:%S %z", "Fri, 20 Feb 2015 22:02:34 UT");
+        assertTimeFormat("%a, %d %b %Y %H:%M:%S %z", "Fri, 20 Feb 2015 22:02:34 GMT");
+        assertTimeFormat(    "%d %b %Y %H:%M:%S %z",      "20 Feb 2015 22:02:34 GMT");
+        assertTimeFormat(    "%d %b %Y %H:%M %z",         "20 Feb 2015 22:02 GMT");
+        assertTimeFormat("%a, %d %b %Y %H:%M %z",    "Fri, 20 Feb 2015 22:02 GMT");
+        assertTimeFormat(    "%d %b %Y",                  "20 Feb 2015");
+        assertTimeFormat("%a, %d %b %Y",             "Fri, 20 Feb 2015");
+        assertTimeFormat("%a, %d %b %Y %H:%M %z",    "Fri, 20 Feb 2015 22:02 +0000");
+        assertTimeFormat("%a, %d %b %Y %H:%M %:z",   "Fri, 20 Feb 2015 22:02 +00:00");
+        assertTimeFormat("%a, %d %b %Y %H:%M %z",    "Fri, 20 Feb 2015 22:02 +00");
+    }
+
+    @Test
+    public void testFormatApacheClf() {
+        assertTimeFormat("%d/%b/%Y:%H:%M:%S %z", "07/Mar/2004:16:05:50 -0800");
+    }
+
+    @Test
+    public void testFormatAnsiCAsctime() {
+        assertTimeFormat("%a %b %e %H:%M:%S %Y", "Fri May 11 21:44:53 2001");
+    }
+
+    @Test
+    public void testMergeMostFrequentMatches() {
+        final TimeFormatMatch[] matches1 = {
+            new FakeMatch("foo"),
+            new FakeMatch("bar"),
+            new FakeMatch("bar"),
+            new FakeMatch("foo"),
+            new FakeMatch("foo"),
+            new FakeMatch("bar"),
+            new FakeMatch("foo"),
+        };
+        assertEquals("foo", TimeFormatGuess.mergeMostFrequentMatches(Arrays.asList(matches1)).getFormat());
+
+        final TimeFormatMatch[] matches2 = {
+            new FakeMatch("bar"),
+            new FakeMatch("foo"),
+            new FakeMatch("bar"),
+            new FakeMatch("bar"),
+            new FakeMatch("foo"),
+            new FakeMatch("bar"),
+            new FakeMatch("foo"),
+            new FakeMatch("foo"),
+            new FakeMatch("bar"),
+        };
+        assertEquals("bar", TimeFormatGuess.mergeMostFrequentMatches(Arrays.asList(matches2)).getFormat());
+    }
+
+    private static class FakeMatch implements TimeFormatMatch {
```

Comment:
why would you need this `FakeMatch` instead of re-using the `SimpleMatch`?

Reply:
The main reason looks to be because `SimpleMatch` is package-private in `org.embulk.util.guess.timeformat`. I thought to it is more resaonable to have `FakeMatch` than to make `SimpleMatch` `public` only for testing.